tool/dart_model_generator/lib/generate_dart_model.dart

```diff
@@ -188,15 +188,22 @@ abstract class Definition {
   /// The name of the defined type.
   String get name;
 
-  /// Defines a class.
+  /// Defines a "class".
+  ///
+  /// Generated as an extension type over `Map<String, Object?>`.
   ///
   /// [createInBuffer] specifies whether the type is written directly to a
   /// buffer when created. If so, it must be created in a `Scope` which will
   /// provide the buffer.
+  ///
+  /// [extraCode] is arbitrary code to write directly in the extension type.
```

Comment:
It might be better if we could add stuff like this via mixin or something?

Reply:
I can't think of any way of getting static methods / constructors ... except augmentations, which we should probably avoid using at least for now.

But very happy to switch to something better if/when it's possible :)

Reply:
ah, didn't realize you were adding statics